### ISSUE TITLE
AdGuard keeps killed by MIUI 12: question AND bug report AND enhancement request

### ISSUE THREAD

**Issue by rhochenburger** [AdguardTeam/AdguardForAndroid#3739]
- **AdGuard version:**
v3.6(1)

- **Filtering mode:**
not relevant

- **Device:**
Xiaomi Mi Max 3

- **Operating system and version:**
Android 10

- **Root access:**
no

since my Xiaomi phone has been upgraded to MIUI Global 12.0.1, AdGuard is killed randomly

the MIUI settings I have applied in vain:
#1 task list > long tap on AdGuard > tap on padlock icon so that it changes to a closed padlock
#2 task list > long tap on AdGuard > tap on "Configure" icon > Battery saver = "No restrictions"
#3 Settings > Modify system settings > Battery optimization > AdGuard > "Don't optimize"

the root cause of the issue seems to be related to #3: this setting is reset back to "Optimize" after some time

actually, I can/could live with this MIUI bug since I have set the watchguard period of AdGuard to 20 secs (so that after being terminated, AdGuard is restarted automatically and almost immediately)

**QUESTION**
is there another relevant MIUI 12 setting I am not aware of?

**ADGUARD BUG REPORT**
1. I set the watchguard period of AdGuard to 20 secs
2. AdGuard is killed by MIUI (no this is still not the AdGuard bug)

_actual behavior:_
3. sometimes (reproducibility: 1 of 10) the restart procedure fails (even after waiting for 1+ min) hence AdGuard needs to be restarted manually

_expected behavior:_
3. the restart procedure shall always succeed

**ADGUARD ENHANCEMENT REQUEST**
1. AdGuard is killed by MIUI
2. AdGuard is restarted either automatically or manually
3. Adguard displays the notification "The battery saving of your system ..."

_actual behavior:_
4. I have to acknowledge the notification manually again and again since there is no config setting (known for me) for disabling these notifications, 

_expected behavior:_
4. there shall be a config setting for disabling these notifications
in other words: please do not punish the user for a MIUI bug

**Comment by Chinaski1** [comment]
Hello there! @rhochenburger 

Unfortunately, there is nothing we can do about this issue on the app side.
You can also try using this instruction: https://dontkillmyapp.com/xiaomi

**Comment by rhochenburger** [comment]
hello @Chinaski1,

could you please comment
- the AdGuard bug report
- the AdGuard enhancement request

above?

**Comment by Chinaski1** [comment]
Watchdog function works in a slightly different way. It reactivates the protection only if the process is running in the system background. In your case, the system unloads the process so Watchdog cannot automatically restart the protection. 

Unfortunately, it is not possible to create the option of restarting the app completely.

To disable this notification go to AdGuard main settings -> **Notifications** -> uncheck **Others**.

**Comment by rhochenburger** [comment]
@Chinaski1 thank you for the answer
